### ISSUE TITLE
Not able to add spark-api and spark-auth-config nodes to palette

### ISSUE THREAD

**Issue by fsiyavud-cisco** [CumberlandGroup/node-red-contrib-spark#1]
I tried to install the nodes through CLI and through the UI, and the process completed both times. The nodes were shown as enabled, but then the UI hangs at load. I disabled the nodes from CLI and enabled them one by one and found that enabling spark-api causes the problem. spark-auth-config gets enabled alright, but it does not get added to the palette.

nodered v0.15.2
node.js v6.9.1
$ node-red-admin info node-red-contrib-spark    Module:    node-red-contrib-spark
Version:   2.0.2

Nodes                                      Types                  State
node-red-contrib-spark/spark-api           Spark API              disabled
node-red-contrib-spark/spark-auth-config   Spark Authentication   enabled
node-red-contrib-spark/spark-parser        Spark Payload Parser   enabled
node-red-contrib-spark/spark-webhook       Spark Webhook          enabled

(Spark-API was disabled my me using CLI so that the UI start working again).

**Comment by nmarus** [comment]
I have heard of similar issues when using node-js 6.x, but have yet been able to replicate it.  

A few questions on your setup if you don't mind. What Operating system (osx, debian, ubuntu, etc) and version? Did you have a previous version (1.x) of this module installed and perhaps a previous flow that node-red is trying to load using the new module? Also, what platform? rasberry-pi? x86-32? x86-64?

**Comment by nmarus** [comment]
Also, the spark-auth module is a config module so you shouldn't see it in the palette. There is only three visible nodes in palette. 

As for the UI hanging but no errors on console, might try a differnt browser? or open the developer pane of Chrome/Firefox/etc and see what the console is logging on the web front end code. That seems to be where the issue is based on what you describe.  

I just tested 2.0.2 in Debian8 and OSX using both node 4.x and 6.x and was able to get everything working.

**Comment by fsiyavud-cisco** [comment]
There is an error in the front-end as you suspected. Here is the log for the case where the UI hanged:

```
Node-RED: 0.15.2
Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience. For more help, check https://xhr.spec.whatwg.org/.
GET https://192.168.0.101/swagger-client-web.js?_=1479183062745 404 (Not Found)
Uncaught SyntaxError: Unexpected identifier(…)

```

I guess this might be an issue with paths. FYI, my node-red node root and admin root are set to specific paths (not at /).

Once I disabled spark-api node, the UI worked with the webook and parser nodes available in the palette. But I got an error on the console even in this case.

```
Node-RED: 0.15.2
GET https://192.168.0.101/webhook/cisco_spark_v1.json 404 (Not Found)

```

BTW, I use Chrome (happens on Safari too, does not matter). node-red is hosted on a RasPI 3 running Raspbian Jessy. I did not have a previous version of the node-red-contrib-spark module installed.

**Comment by nmarus** [comment]
The problem happens because your node-red root and admin root paths are non default. I should be able to implement a fix that accounts for that sometime this week. Thanks for the feedback. I'll follow up here when i have a fix in place.

**Comment by fsiyavud-cisco** [comment]
Sure. Thanks.

**Comment by nmarus** [comment]
This issue is still happening for those that have modified their node-red setting from using the non-default roots for admin and user. 

Details: 
- This node-red module creates a static binding admin binding under `http://myred:port/swagger-client-web.js`
  - If this URL is not accessible, or if you have changed the `httpAdminRoot` or `httpRoot` in your node-red settings, you will not see anything listed under the 'Resource' or 'Method' drop-downs of the API Node.
- When using the webhook node, this node-red module creates a admin binding under `http://myred:myport/webhook/cisco_spark_v1.json`
  - If this URL is not accessible, or if you have changed the `httpAdminRoot` or `httpRoot` in your node-red settings, you will not see anything listed under the 'Resource' drop-downs of the Webhook Node.
- When using the webhook node, this node-red module creates a user binding under `http://myred:myport/spark<uuid>`
  - The exact webhook url will be posted to the node-red console if you wish to test that it responds to POST request. 
  - If this URL is not accessible, or if you have changed the `httpNodeRoot` or `httpRoot` in your node-red settings, you will not receive any inbound hooks from the Spark API.

**Comment by nmarus** [comment]
Try fix applied in v2.0.5. Customized httpNodeRoot and httpAdminRoot properties in the settings.js for Node-Red should not break this anymore. If it still does, please open a new issue with the details.
